plugins/modules/cloud/scaleway/scaleway_container_namespace.py

```diff
@@ -0,0 +1,286 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Scaleway Serverless container namespace management module
+#
+# Copyright (c) 2022, Guillaume MARTINEZ <lunik@tiwabbit.fr>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: scaleway_container_namespace
+short_description: Scaleway Container namespace management module
+version_added: 5.8.0
+author: Guillaume MARTINEZ (@Lunik)
+description:
+  - This module manages container namespaces on Scaleway account.
+extends_documentation_fragment:
+  - community.general.scaleway
+  - community.general.scaleway_waitable_resource
+
+
+options:
+  state:
+    type: str
+    description:
+      - Indicate desired state of the container namespace.
+    default: present
+    choices:
+      - present
+      - absent
+
+  project_id:
+    type: str
+    description:
+      - Project identifier.
+    required: true
+
+  region:
+    type: str
+    description:
+      - Scaleway region to use (for example C(fr-par)).
+    required: true
+    choices:
+      - fr-par
+      - nl-ams
+      - pl-waw
+
+  name:
+    type: str
+    description:
+      - Name of the container namespace.
+    required: true
+
+  description:
+    description:
+      - Description of the container namespace.
+    type: str
+
+  environment_variables:
+    description:
+      - Environment variables of the container namespace.
+      - Injected in containers at runtime.
+    type: dict
+
+  secret_environment_variables:
+    description:
+      - Secret environment variables of the container namespace.
+      - Updating thoses values will not output a C(changed) state in Ansible.
```

Comment:
Same comment as in https://github.com/ansible-collections/community.general/pull/5415#discussion_r1002756112.

From the definition in https://github.com/ansible-collections/community.general/blob/main/plugins/module_utils/scaleway.py#L83-L92 it seems that values from this option are only updated if there are also other changes. This sounds wrong to me.

Reply:
I agree with you on the topic. I have tried to explain my choice on the matter on the previous PR : https://github.com/ansible-collections/community.general/pull/5399#discussion_r1001579373

tl;dr Some provided field (`secret_environment_variables` in that case) are returned with a different format by the API. There is no way to check if the value must be updated or not.